### DISCUSSION TITLE
YAML config question

### DISCUSSION THREAD

**Discussion by AKMourato** [facebookresearch/detectron2#3423] [Q&A]
Greetings.
setting these parameters:
```
cfg.INPUT.MIN_SIZE_TRAIN = 1024
cfg.INPUT.MIN_SIZE_TEST = 1024
cfg.INPUT.MAX_SIZE_TRAIN = 4096
cfg.INPUT.MAX_SIZE_TEST = 4096
```
does the image aspect ratio is maintained when the dataloader sampler chooses one specific size, or in the extreme is it possible that some images will have the short_edge_length = 1024 and the other dimension = 4096?

**Comment by AKMourato** [comment]
hi @ppwwyyxx, since you apparently don't come here I took the freedom to tag you sorry.
With these settings, do we have image dimension ranges like: (1024, 2048), (4096,1024), (2048,2048), (2048, 4096)?

Also, I can't find a proper explanation of ROI_HEADS.BATCH_SIZE_PER_IMAGE anywhere. FasterR-CNN and MaskR-CNN papers don't specifically mention it.
Having 1000 proposals per image coming from RPN (standard setting), and considering an image batch size of 1, what exactly this parameters does in your implementation?
Thanks in advance.

**Comment by juanluisrosaramos** [comment]
If you have a look at the default datamapper in the line         transforms = self.augmentations(aug_input)
you have the scale transformations from cfg.INPUT.MIN_SIZE_TRAIN[] in my case for an image of 616*512 datectron2 is scaling to 847*704 before the step training, so it is maintaining aspect ratio.
TransformList[ResizeTransform(h=512, w=616, new_h=704, new_w=847, interp=2)]

**Comment by AKMourato** [comment]
@ppwwyyxx check this when you have time please

**Comment by ppwwyyxx** [comment]
https://detectron2.readthedocs.io/en/latest/modules/data_transforms.html#detectron2.data.transforms.ResizeShortestEdge is used by default and explains how resize is computed.

https://detectron2.readthedocs.io/en/latest/modules/config.html#yaml-config-references explains ROI_HEADS.BATCH_SIZE_PER_IMAGE

**Comment by AKMourato** [comment]
thanks @ppwwyyxx, but I already check that, though still not clarified.
1) my doubt is, if the ResizeShortEdge has "choice" sampler, short_edge = 1024 and max_size = 4096, for original inputs of 4096x4096, what are the possible images sizes feed to the network? Can you specify please?

2) the comments are not that explanatory enough, and in the original papers that parameter is not specified either. 
If the Box Head is being fed with 1000 proposals coming from RPN per image (considering batch_size=1), the number of regions of interest head will do exactly what?
